### PR TITLE
fix(editor): Move versions check to init function and refactor store (no-changelog)

### PR DESCRIPTION
**Pull request by alexgrozav**
After recent app initialization changes, the version check was running before the settings loaded.

This PR fixes loading, refactors the newVersions mixin, and adds required tests.

https://github.com/n8n-io/n8n/issues/8058

## Review / Merge checklist
- [x] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [x] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

**Review by mutdmour — CHANGES_REQUESTED** [review]
Nice, thanks for fixing. Could you please add an e2e that updates panel opens and versions load?

**Comment by alexgrozav** [comment]
> Nice, thanks for fixing. Could you please add an e2e that updates panel opens and versions load?

Added e2e tests. Please review 🙂

File: cypress/e2e/36-versions.cy.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		cy.intercept('GET', '/rest/settings', (req) => {
+			req.continue((res) => {
+				if (res.body.hasOwnProperty('data')) {
+					res.body.data = {
+						...res.body.data,
+						releaseChannel: 'stable',
+						versionCli: '1.0.0',
+						versionNotifications: {
+							enabled: true,
+							endpoint: 'https://api.n8n.io/api/versions/',
```
Comment (mutdmour) [inline cypress/e2e/36-versions.cy.ts:22]: should not we mock the response from this endpoint?

Comment (alexgrozav) [reply cypress/e2e/36-versions.cy.ts:23]: Good point! Done.

**Review by mutdmour — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.22.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.22.0)

**Pull request merged**